### PR TITLE
R4R: Refactor denom&nft logic

### PR DESCRIPTION
**Pull request by zhiqiang-bianjie**

### CODE REVIEW

File: keeper/denom.go [hunk 1]
```diff
@@ -7,31 +7,47 @@ import (
...
+
+	if k.HasDenomNm(ctx, denom.Name) {
+		return sdkerrors.Wrapf(types.ErrInvalidDenom, "denomName %s has already exists", denom.Name)
 	}
 
 	store := ctx.KVStore(k.storeKey)
 	bz := k.cdc.MustMarshalBinaryBare(&denom)
-	store.Set(types.KeyDenom(denom.Name), bz)
+	store.Set(types.KeyDenomID(denom.ID), bz)
+	if len(denom.Name) == 0 {
```
Comment (chengwenxi) [inline keeper/denom.go:38]:
```suggestion
	if len(denom.Name) > 0 {
```

File: types/denom.go [hunk 2]
```diff
@@ -9,29 +9,32 @@ import (
...
 	return nil
 }
 
 func (d Denom) String() string {
-	return fmt.Sprintf(`Name:				%s
+	return fmt.Sprintf(`ID:				%s
+Name:			%s
 Schema:			%s
 Creator:		%s`,
+		d.ID,
```
Comment (chengwenxi) [inline types/denom.go:37]: fmt

File: keeper/denom.go (hunk 1 above)
Comment (chengwenxi) [reply keeper/denom.go:38]: if deom name is empty, should delete it.

File: client/cli/flags.go [hunk 3]
```diff
@@ -23,19 +26,23 @@ var (
...
 	FsMintNFT.String(FlagTokenURI, "", "URI for supplemental off-chain tokenData (should return a JSON object)")
 	FsMintNFT.String(FlagRecipient, "", "Receiver of the nft, if not filled, the default is the sender of the transaction")
-	FsMintNFT.String(FlagTokenData, "", "the origin data of nft")
+	FsMintNFT.String(FlagTokenData, "", "The origin data of nft")
+	FsMintNFT.String(FlagTokenName, "", "The name of nft")
 
 	FsEditNFT.String(FlagTokenURI, "[do-not-modify]", "URI for supplemental off-chain tokenData (should return a JSON object)")
-	FsEditNFT.String(FlagTokenData, "[do-not-modify]", "the tokenData of nft")
+	FsEditNFT.String(FlagTokenData, "[do-not-modify]", "The tokenData of nft")
+	FsEditNFT.String(FlagTokenName, "", "The name of nft")
```
Comment (chengwenxi) [inline client/cli/flags.go:39]: default value should be "[do-not-modify]"

File: types/denom.go (hunk 2 above)
Comment (zhiqiang-bianjie) [reply types/denom.go:37]: no need

File: keeper/denom.go (hunk 1 above)
Comment (zhiqiang-bianjie) [reply keeper/denom.go]: not be updated

**Pull request merged**